src/rs.cpp

```diff
@@ -679,7 +729,15 @@ HANDLE_EXCEPTIONS_AND_RETURN(, options, option, value)
 rs2_options_list* rs2_get_options_list(const rs2_options* options, rs2_error** error) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL(options);
-    return new rs2_options_list{ options->options->get_supported_options() };
+    auto rs2_list = new rs2_options_list;
+    auto option_ids = options->options->get_supported_options();
+    rs2_list->list.reserve( option_ids.size() );
+    for( auto option_id : option_ids )
+    {
+        auto wrapper = new rs2_option_value_wrapper( option_id, {} );  // empty json
```

Comment:
Can we allocate an array of option_wrappers to avoid multiple allocations?
Not sure the list will be deleted as a whole, then it will be problematic releasing.

Reply:
Not really -- each has a lifetime of its own.